### PR TITLE
Dynamic enable/disable GestureNavContract

### LINKED ISSUE #3762: [FEATURE] Use SEPARATE-RECENT-ACTIVITIE which will add closing animation for apps
### Describe the feature

In launcher3 i found code which has separate recent activity which adds the closing animations for custom launchers adding this feature would improve launcher experience and using closing animations without quick switch

### Why would it be useful to add?

The launcher experience would be enhanced for non rooted users

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

Screenshot_20231124-193845
This can be found in developer options if possible

### Additional context

Maybe this feature was used in launchers such as hyperion and nova this should work with android 12 devices but not with i think android 11 lower

### LINKED ISSUE #3784: [FEATURE] `GestureNavContract` support
### Describe the feature

Technically a dupe of #2856, although it was closed as stale a while back and I thought I might open an issue focusing a more specific and realistic implementation for rootless users.

Since Android 11 or so, many launcher3-based system launchers and some OEMs such as Samsung have offered the [`GestureNavContract`](https://cs.android.com/android/platform/superproject/+/master:packages/apps/Launcher3/src/com/android/launcher3/GestureNavContract.java) class to third party launchers as a method of providing better animations to users without the need for something like QuickSwitch.

While not being exactly the same, the animations are an improvement over the current "scale-to-the-middle" effect that apps do by default. Lawnchair currently doesn't seem to use this API, although other launchers such as nova, hyperion, and niagra have implemented animations using this API to [pretty good success](https://twitter.com/prjkt_io/status/1551634957030010881).

### Why would it be useful to add?

Offers a better experience/opening and closing animations for rootless users and those who do not use quickswitch.

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

https://github.com/LawnchairLauncher/lawnchair/assets/42101043/978662bf-79f7-4a97-a545-fb1386818b06

https://twitter.com/prjkt_io/status/1551634957030010881

### Additional context

Before A13, there was a [bug](https://issuetracker.google.com/issues/202864574) where the animations tended to end early, although this has been fixed in AOSP and by many OEMs following Android 13's release.

### PR DESCRIPTION
**Pull request by MrSluffy**
## Description

Introducing this enable/disable dynamically `GestureNavContract` feature provides  non-rooted users with the opportunity to enjoy a sophisticated gesture-based app-to-home navigation experience through the `GestureNavContract` without breaking for Quickswitch users.

Demo 

https://github.com/LawnchairLauncher/lawnchair/assets/36076410/5dba1623-b6bb-43ff-a4ef-e91e813c8b58

Closes #3784 and #3762

## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lawnchair/res/values/strings.xml [hunk 1]
```diff
@@ -60,6 +60,8 @@
     <string name="transparent_background_icons">Transparent Themed Icons</string>
     <string name="transparent_background_icons_description">Use transparent background on themed icons</string>
     <string name="smartspace_mode_selection">At a Glance Provider selection</string>
+    <string name="pref_separate_recents_activity_title">Separate Recents Activity</string>
```
Comment (SuperDragonXD) [inline lawnchair/res/values/strings.xml]: Switch preference options should ideally be in `<verb>` `<what to do>`. Also no one will need to know the internal code term, so I've clarified it as well by using "GestureNavContract":

```suggestion
    <string name="pref_separate_recents_activity_title">Enable GestureNavContract</string>
```

File: lawnchair/res/values/strings.xml [hunk 2]
```diff
@@ -60,6 +60,8 @@
...
+    <string name="pref_separate_recents_activity_description">Uses a separate recents activity instead of using the integrated recents+Launcher UI (GesturNavContract API)</string>
```
Comment (SuperDragonXD) [inline lawnchair/res/values/strings.xml]: Users will have no idea what GestureNavContract means, so I've also clarified the description.
```suggestion
    <string name="pref_separate_recents_activity_description">Improves app animations by interfacing with the system launcher. Not all OEMs support it.</string>
```

**Review by SuperDragonXD — CHANGES_REQUESTED** [review]
Looks fine, although I'd suggest clarifying the wording.

I'd also recommend hiding the option if QuickSwitch is available (via adding a `LawnchairApp.isRecentsAvailable` check).

**Comment by suphon-t** [comment]
Can't we just change the condition [here](https://github.com/LawnchairLauncher/lawnchair/blob/f696cf85556ef1f875afef9bd8b250ed80ac8442/quickstep/src/com/android/launcher3/uioverrides/QuickstepLauncher.java#L628) instead of adding a toggle? I feel like that would be much cleaner, and would work without the user changing any settings.

**Comment by MrSluffy** [comment]
> Can't we just change the condition [here](https://github.com/LawnchairLauncher/lawnchair/blob/f696cf85556ef1f875afef9bd8b250ed80ac8442/quickstep/src/com/android/launcher3/uioverrides/QuickstepLauncher.java#L628) instead of adding a toggle? I feel like that would be much cleaner, and would work without the user changing any settings.

Let me try

File: lawnchair/res/values/config.xml [hunk 3]
```diff
@@ -104,6 +104,7 @@
     <bool name="config_default_show_icon_labels_in_drawer">true</bool>
     <bool name="config_default_enable_fuzzy_search">false</bool>
     <bool name="config_default_enable_smartspace">true</bool>
+    <bool name="config_default_separate_recent_activity">false</bool>
```
Comment (SuperDragonXD) [inline lawnchair/res/values/config.xml]: Do we still need this? I think this can be safely removed.

**Review by SuperDragonXD — APPROVED** [review]

File: lawnchair/res/values/config.xml (hunk 3 above)
Comment (MrSluffy) [reply lawnchair/res/values/config.xml]: yeah let me remove it

**Review by suphon-t — APPROVED** [review]

**Pull request merged**